test/replaytests/workflows.go

```diff
@@ -217,3 +217,39 @@ func SideEffectWorkflow(ctx workflow.Context, field string) error {
 func EmptyWorkflow(ctx workflow.Context, _ string) error {
 	return nil
 }
+
+func MutableSideEffectWorkflow(ctx workflow.Context) ([]int, error) {
```

Comment:
Is it possible, maybe in addition to these replay tests, to replicate the error that caused this issue in the first place? Maybe this workflow will, haven't tested, but want to make sure we can verify new behavior is for new workflow run against server.

Also I wonder if we want to have an integration test proving non-deterministic mutable side effect order change. That may be asking too much though.

Reply:
> Or is that what that small assertion addition in workflow_test.go already does?

Yes exactly

>Also I wonder if we want to have an integration test proving non-deterministic mutable side effect order change. That may be asking too much though.

not sure I understand, do you mean a test that shows changing the order of mutable side effects is not deterministic?

Reply:
> do you mean a test that shows changing the order of mutable side effects is not deterministic?

I do. Basically if I add a mutable side effect for the ID if it's non-deterministic. But I think this is a bit much, no need.